src/gh-slugify.js

```diff
@@ -46,24 +46,67 @@ export default class GithubSlugify {
       .replace(whitespace, this.replacementChar);
   }
 
+  /**
+   * Helper function to extract text from a node.
+   * @api public
+   * @param {React.ReactElement} node - the react element to extract text from.
+   * @returns {String} the node text extracted from the node.
+   */
+  extractString(node) {
+    let title = '';
+    if (node.props && node.props.children) {
+      if (typeof node.props.children === 'object') {
+        node.props.children.forEach((child) => {
+          title += this.extractString(child);
+        });
+      } else {
+        title += node.props.children;
+      }
+    } else {
+      title += node;
+    }
+    return title;
+  }
+
   /**
    * Generates a GH style slug from the passed text.
    * @api public
    * @param {String} text - the txt to be converted to a slug.
    * @returns {String} the text converted to a slug.
    */
   slug(text) {
-    const slug = this.replace(text);
+    let slug = this.replace(text);
+
+    // Prepend slug with `_` when it begins with a number
+    if (!isNaN(slug.replace('#', '').charAt(0))) {
+      slug = '_' + slug.replace('#', '');
+    }
```

Comment:
Why make this change?

Reply:
We implemented this locally in our own codebase to support slugs that begin with numbers. Programmatically, there is no issue, but when it is used in a browser, browsers don't handle hashes prepended with numbers. I thought this would be a good addition to aid others as well.

Reply:
After investigating further, I realized this is specific to our use case. [As indicated here](https://stackoverflow.com/a/37271406), a hash can start with numbers (or even only numbers) and the browser allows it. However, for our use case, we were using smooth scrolling, which uses CSS selectors, which _do not_ allow for numbers to start a queryselector. So I will remove this logic and keep it in our codebase.